### ISSUE TITLE
HEAD request on hystrix.stream is not handled correctly

### ISSUE THREAD

**Issue by joshiste** [Netflix/Hystrix#1369]
When I issue subsequent HEAD requests (via `curl -I`) to `/hystrem.stream` I get a 503 response on the 6th request ("MaxConcurrentConnections reached: 5"). 

Also in the logs the message for stopping the HystrixMetricsPoller is missing.

Log from 2 HEAD requests:
```
2016-10-01 21:34:42.857  INFO 1 --- [p-nio-80-exec-2] c.n.h.c.m.e.HystrixMetricsPoller         : Starting HystrixMetricsPoller
2016-10-01 21:34:42.857  INFO 1 --- [p-nio-80-exec-2] c.n.h.c.m.e.HystrixMetricsStreamServlet  : Starting poller
2016-10-01 21:34:43.994  INFO 1 --- [p-nio-80-exec-3] c.n.h.c.m.e.HystrixMetricsPoller         : Starting HystrixMetricsPoller
2016-10-01 21:34:43.994  INFO 1 --- [p-nio-80-exec-3] c.n.h.c.m.e.HystrixMetricsStreamServlet  : Starting poller
```

Log from 2 GET requests:

```
2016-10-01 21:34:20.888  INFO 1 --- [-nio-80-exec-10] c.n.h.c.m.e.HystrixMetricsPoller         : Starting HystrixMetricsPoller
2016-10-01 21:34:20.888  INFO 1 --- [-nio-80-exec-10] c.n.h.c.m.e.HystrixMetricsStreamServlet  : Starting poller
2016-10-01 21:34:22.891  INFO 1 --- [-nio-80-exec-10] c.n.h.c.m.e.HystrixMetricsPoller         : Stopping the HystrixMetricsPoller
2016-10-01 21:34:22.965  INFO 1 --- [p-nio-80-exec-1] c.n.h.c.m.e.HystrixMetricsPoller         : Starting HystrixMetricsPoller
2016-10-01 21:34:22.965  INFO 1 --- [p-nio-80-exec-1] c.n.h.c.m.e.HystrixMetricsStreamServlet  : Starting poller
2016-10-01 21:34:25.469  INFO 1 --- [p-nio-80-exec-1] c.n.h.c.m.e.HystrixMetricsPoller         : Stopping the HystrixMetricsPoller
```

I guess the concurrent connection handling isn't correct for HEAD requests.

**Comment by mukteshkrmishra** [comment]
It happens because hystrix stream is a continuous stream not a regular servlet. You cannnot consume it via direct curl. Each time when you tries to curl it opens a connection but due to continuous stream it does. It get closed. There is a default limit of 5 connection is established as of now that is why you are getting this error, after 5 request.

**Comment by joshiste** [comment]
By doing subsequent regular GETs, aborting the current before starting the next, I don't hit the limit.

**Comment by mukteshkrmishra** [comment]
How are you so sure that your request is aborted successfully. The way I understand is, it is not possible to curl hystrix SSE stream directly.

**Comment by joshiste** [comment]
Just hitting ctrl-c ... Which terminates curl and closes the underlying TCP connection....

**Comment by mattrjacobs** [comment]
Thanks for the report, @joshiste .  I took a glance at the code and we do indeed only perform connection management on HTTP GETs.  What's your use case for preferring HTTP HEAD?

**Comment by joshiste** [comment]
I'm the maintainer of https://github.com/codecentric/spring-boot-admin and this I wanted to check if a registered boot app has a hystrix.stream-endpoint.
My current solution is to do a GET and abort after receiving the first chunk, so it's ok for me if you don't solve this issue, 

But imho you should handle HEAD properly or return a 405; The current situation just ends in the unavailability of the endpoint and is somehow disappointing...

**Comment by mattrjacobs** [comment]
Yeah, that all makes sense and I agree.  Will get it in the backlog, that sounds fairly straightforward to get cleaned up.
